### PR TITLE
Feat: environment resource does not support new templateless creation

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

This is a large/complex change. The feature will be divided into small PRs.
In this PR I've moved the template schema to a common function that receives template type (shared vs single).

part of #374

### CODE REVIEW

File: env0/templates.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+		}
+
+		return ret
+	}
+
+	s := map[string]*schema.Schema{
+		"name": {
+			Type:        schema.TypeString,
+			Description: "name to give the template",
+			Required:    templateType == TemplateTypeShared,
```
Comment (TomerHeber) [inline env0/templates.go:55]: varying behavior...

File: env0/templates.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+			Computed:    templateType == TemplateTypeSingle,
+		},
+		"description": {
+			Type:        schema.TypeString,
+			Description: "description for the template",
+			Optional:    true,
+			Computed:    templateType == TemplateTypeSingle,
```
Comment (TomerHeber) [inline env0/templates.go:62]: varying behavior...

File: env0/templates.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+			Type:          schema.TypeBool,
+			Description:   "true if this template uses github enterprise repository",
+			Optional:      true,
+			Default:       "false",
+			ConflictsWith: allVCSAttributesBut("is_github_enterprise"),
+		},
+	}
+
+	if templateType == TemplateTypeSingle {
+		s["id"] = &schema.Schema{
```
Comment (TomerHeber) [inline env0/templates.go:180]: varying behavior...
need to store the template id.

**Review by sagilaufer1992 — APPROVED** [review]
LGTM

**Pull request merged**